### PR TITLE
Feature/open update cfd fixed

### PR DESCRIPTION
**Pull request by arealclimber**
### DEVELOP

- [x]  OpenSubTab
- [x] PositionOpenModal
- [x] WalletPanel
- [x] OpenPositionItem
- [x] UpdateFormModal
- [x] userContext

### CHECK LIST

- [Naming convention](https://github.com/CAFECA-IO/WorkGuidelines/blob/main/technology/coding-convention/naming-convention.md) verification: checked
- Coding style verification: checked
- new Library: 0
- new Class / Component: 0
- new loop: 0
- new recursive: 0
- high risk: 0
- new sql: 0

### UML

- [Sequence Diagram 001 (interaction between user, frontend and backend)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00001.md)
- [Sequence Diagram 002 (background mechanism)](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDSD00002.md)
- [Component Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCP00001.md)
- [Class Diagram](https://github.com/CAFECA-IO/Documents/blob/main/TBD/TBDCD00001.md)
- [Global context document](https://www.notion.so/Global-Context-public-d93ac7c941c94fb6b173d34a7bc4e959)
- [Table of interface](https://github.com/CAFECA-IO/EventReport/blob/main/meeting/20230106_tbd_mvp.md)

### CODE REVIEW

File: src/components/position_open_modal/position_open_modal.tsx [hunk 1]
```diff
@@ -71,7 +71,7 @@ const PositionOpenModal = ({
     const order: IApplyCreateCFDOrder = {
       ...openCfdRequest,
       createTimestamp: getTimestamp(),
-      liquidationTime: getTimestamp() + 86400,
+      liquidationTime: openCfdRequest.quotation.deadline + 86400,
```
Comment (arealclimber) [inline src/components/position_open_modal/position_open_modal.tsx:74]: 86400 應該是常數

File: src/components/open_sub_tab/open_sub_tab.tsx [hunk 2]
```diff
@@ -1,23 +1,185 @@
...
+      if (!longQuotationRef.current || !shortQuotationRef.current) return;
+
+      const base = longQuotationRef.current.deadline - WAITING_TIME_FOR_USER_SIGNING;
+      const tickingSec = (base * 1000 - getTimestampInMilliseconds()) / 1000;
+      setSecondsLeft(tickingSec > 0 ? Math.round(tickingSec) : 0);
+
+      if (secondsLeftRef.current === 0) {
+        setQuotation();
+      }
+    }, 1000);
```
Comment (arealclimber) [inline src/components/open_sub_tab/open_sub_tab.tsx:164]: setTimeout

File: src/contexts/user_context.tsx [hunk 3]
```diff
@@ -702,8 +702,11 @@ export const UserProvider = ({children}: IUserProvider) => {
   */
 
   const updateBalance = (updatedBalance: IBalance) => {
+    // Deprecated: not found currency (20230430 - Shirley)
+    // eslint-disable-next-line no-console
+    console.log('arg in updateBalance in ctx', updatedBalance);
     if (balancesRef.current) {
-      const index = balancesRef.current?.findIndex(balance => balance.currency);
+      const index = balancesRef.current?.findIndex(balance => balance?.currency);
```
Comment (arealclimber) [inline src/contexts/user_context.tsx:709]: - try-catch

Comment (arealclimber) [reply src/contexts/user_context.tsx:709]: 用到 `updateBalance` 的地方也要 try-catch

File: src/components/open_sub_tab/open_sub_tab.tsx (hunk 2 above)
Comment (arealclimber) [reply src/components/open_sub_tab/open_sub_tab.tsx:207]: - 跟著折線圖最後一個值+點差

Comment (arealclimber) [reply src/components/open_sub_tab/open_sub_tab.tsx:207]: #565

File: src/contexts/user_context.tsx [hunk 4]
```diff
@@ -754,14 +754,25 @@ export const UserProvider = ({children}: IUserProvider) => {
...
+      if (balancesRef.current) {
+        const index = balancesRef.current?.findIndex(balance => balance.currency);
+        if (index !== -1) {
+          const updateBalances = [...balancesRef.current];
+          updateBalances[index] = updatedBalance;
+          setBalances(updateBalances);
+        } else throw new CustomError(Code.BALANCE_NOT_FOUND);
       } else throw new CustomError(Code.BALANCE_NOT_FOUND);
-    } else throw new CustomError(Code.BALANCE_NOT_FOUND);
+    } catch (error) {
```
Comment (arealclimber) [inline src/contexts/user_context.tsx:772]: - 如果`setBalances(updateBalances);`出錯，這邊丟的是 `throw new CustomError(Code.BALANCE_NOT_FOUND)` 錯誤

**Review by Luphia — APPROVED** [review]
- no bug found

**Pull request merged**